pkg/bootstrap/testdata/running_golden.json

```diff
@@ -186,6 +190,30 @@
       }
     }
   },
+  "overload_manager": {
+    "refresh_interval": "1s",
+    "actions": [
+      {
+        "name": "envoy.overload_actions.shrink_heap",
```

Comment:
Given there is no garbage collection in c++, how does envoy shrink heap?

Reply:
https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/overload_manager/overload_manager

"Envoy will periodically try to shrink the heap by releasing free memory to the system"

I am not sure this is valid, as glibc will have used brk() to extend the heap size already. i don't know if there is a counter-action to brk().

I think the more important problem is envoy is OOMing in production and not honoring limits set by Kubernetes (which turn into system limits in the container).

A graceful overload to not accepting requests is better than ungracefully killing all of them.

I do not claim to know how this overload manager is implemented however.

Cheers,
-steve

Reply:
@howardjohn was this closed in favor of another solution? We're seeing our `istio-ingressgateway` pods being OOMKilled on our higher-traffic clusters, and we'd like to see if `overload_manager` would give us some relief. Is this (or an alternative) exposed by `istio`?